File: problem2.js
/* Project Euler Sum of even Fibonacci 
sequence not exceding 4000000 in the sequence */

function fibSequence(limit) {
	var prev = 1;
	var current = 1;
	var fib = 0;
	while (current < limit) {
		if ((prev + current) % 2 == 0 
			&& prev + current <= limit) {
			fib += (prev + current);
		}

		old_current = current;
		current = (prev + current);
		prev = old_current;
	}

	return fib;
}

console.log(fibSequence(4000000));

File: problem2.rb
# Project Euler Sum of even Fibonacci 
# sequence not exceding 4000000 in the sequence

def fib(limit)
	fib = 0
	prev = 1
	current = 1
	while current < limit
		if (prev + current).even? && (current + prev) < limit
			fib += (prev + current)
		end

		old_current = current
		current = prev + current
		prev = old_current
	end
	fib
end

puts fib(4000000)